File: Yamore/Yamore.Model/WeatherForecast.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Yamore.Model
{
    public class WeatherForecast
    {
        public int ForecastId { get; set; }

        public int RouteId { get; set; }

        public DateTime? ForecastDate { get; set; }

        public decimal? Temperature { get; set; }

        public string? Condition { get; set; }

        public decimal? WindSpeed { get; set; }

        public virtual Route Route { get; set; } = null!;
    }
}

File: Yamore/UI/yamore_mobile/lib/models/weather_forecast.dart
class WeatherForecastModel {
  final int forecastId;
  final int routeId;
  final DateTime? forecastDate;
  final double? temperature;
  final String? condition;
  final double? windSpeed;

  WeatherForecastModel({
    required this.forecastId,
    required this.routeId,
    this.forecastDate,
    this.temperature,
    this.condition,
    this.windSpeed,
  });

  factory WeatherForecastModel.fromJson(Map<String, dynamic> json) {
    return WeatherForecastModel(
      forecastId: json['forecastId'] as int,
      routeId: json['routeId'] as int,
      forecastDate: json['forecastDate'] != null
          ? DateTime.tryParse(json['forecastDate'] as String)
          : null,
      temperature: (json['temperature'] as num?)?.toDouble(),
      condition: json['condition'] as String?,
      windSpeed: (json['windSpeed'] as num?)?.toDouble(),
    );
  }
}

